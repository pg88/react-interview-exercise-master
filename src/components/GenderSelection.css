:local(.genderContainer) {
    width: 100%;
    display: block;
}
:local(.genderInputs) {
    display: inline-block;
    margin: 5px;
}