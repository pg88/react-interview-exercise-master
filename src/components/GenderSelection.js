import React, { Component } from 'react';
import { connect } from 'react-redux';
import styles from './GenderSelection.css';
import { FEMALE, GENDER } from '../constants/ActionTypes';


class GenderSelection extends Component {
    constructor(props, context) {
        super(props, context);
        this.state = {
            gender: FEMALE
        }
    }

    selectGender(e) {
        const selectedGender = e.target.value;
        this.setState({
            gender: selectedGender
        });
        this.props.onSelectGender(selectedGender)
    }

    displayGenders() {
        return GENDER.map((value, key) => {
            return (<div key={ value } className={ styles.genderInputs }>
                <input
                    type='radio'
                    name='gender'
                    value={ value }
                    tabIndex={ key }
                    onChange={ this.selectGender.bind(this) }
                /> { value }
            </div>)
        })
    }

    render () {
        return(
            <div className={ styles.genderContainer }>
                <div>
                    <span><b>Gender:</b></span>
                </div>
                { this.displayGenders() }
            </div>
        )
    }
}

export default GenderSelection