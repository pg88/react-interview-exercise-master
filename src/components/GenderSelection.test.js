import React from 'react';
import { mount, shallow } from 'enzyme';
import GenderSelection from './GenderSelection';
import { OTHER, FEMALE } from '../constants/ActionTypes';

describe('GenderSelection tests', () => {
    const selectFunction = jest.fn();
    it('should render correctly', () => {
        const component = shallow(<GenderSelection/>);
        expect(component).toMatchSnapshot();
    });
    it('should select some gender from the available', () => {
        const component = shallow(<GenderSelection/>);
        component.find(`#${ FEMALE }`).simulate('click');
        component.setState({ gender: FEMALE });
        expect(selectFunction.mock.calls.length).toBeDefined();
        expect(component.state().gender).toBeDefined()
    });
});