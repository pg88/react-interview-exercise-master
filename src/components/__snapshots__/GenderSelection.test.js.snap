exports[`GenderSelection tests should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GenderSelection />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className={undefined}>
          <span>
                    Gender:
          </span>
</div>,
        Array [
          <div
            className={undefined}>
            <input
                        checked={false}
                        id="Female"
                        name="gender"
                        onChange={[Function]}
                        tabIndex={0}
                        type="radio"
                        value="Female" />
             
            Female
</div>,
          <div
            className={undefined}>
            <input
                        checked={false}
                        id="Male"
                        name="gender"
                        onChange={[Function]}
                        tabIndex={1}
                        type="radio"
                        value="Male" />
             
            Male
</div>,
          <div
            className={undefined}>
            <input
                        checked={false}
                        id="Other"
                        name="gender"
                        onChange={[Function]}
                        tabIndex={2}
                        type="radio"
                        value="Other" />
             
            Other
</div>,
        ],
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span>
            Gender:
</span>,
          "className": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Gender:",
          },
          "ref": null,
          "rendered": "Gender:",
          "type": "span",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "Female",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              checked={false}
              id="Female"
              name="gender"
              onChange={[Function]}
              tabIndex={0}
              type="radio"
              value="Female" />,
            " ",
            "Female",
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "checked": false,
              "id": "Female",
              "name": "gender",
              "onChange": [Function],
              "tabIndex": 0,
              "type": "radio",
              "value": "Female",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          " ",
          "Female",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "Male",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              checked={false}
              id="Male"
              name="gender"
              onChange={[Function]}
              tabIndex={1}
              type="radio"
              value="Male" />,
            " ",
            "Male",
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "checked": false,
              "id": "Male",
              "name": "gender",
              "onChange": [Function],
              "tabIndex": 1,
              "type": "radio",
              "value": "Male",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          " ",
          "Male",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "Other",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              checked={false}
              id="Other"
              name="gender"
              onChange={[Function]}
              tabIndex={2}
              type="radio"
              value="Other" />,
            " ",
            "Other",
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "checked": false,
              "id": "Other",
              "name": "gender",
              "onChange": [Function],
              "tabIndex": 2,
              "type": "radio",
              "value": "Other",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          " ",
          "Other",
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className={undefined}>
            <span>
                        Gender:
            </span>
</div>,
          Array [
            <div
              className={undefined}>
              <input
                            checked={false}
                            id="Female"
                            name="gender"
                            onChange={[Function]}
                            tabIndex={0}
                            type="radio"
                            value="Female" />
               
              Female
</div>,
            <div
              className={undefined}>
              <input
                            checked={false}
                            id="Male"
                            name="gender"
                            onChange={[Function]}
                            tabIndex={1}
                            type="radio"
                            value="Male" />
               
              Male
</div>,
            <div
              className={undefined}>
              <input
                            checked={false}
                            id="Other"
                            name="gender"
                            onChange={[Function]}
                            tabIndex={2}
                            type="radio"
                            value="Other" />
               
              Other
</div>,
          ],
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span>
              Gender:
</span>,
            "className": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Gender:",
            },
            "ref": null,
            "rendered": "Gender:",
            "type": "span",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "Female",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                checked={false}
                id="Female"
                name="gender"
                onChange={[Function]}
                tabIndex={0}
                type="radio"
                value="Female" />,
              " ",
              "Female",
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "checked": false,
                "id": "Female",
                "name": "gender",
                "onChange": [Function],
                "tabIndex": 0,
                "type": "radio",
                "value": "Female",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            " ",
            "Female",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "Male",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                checked={false}
                id="Male"
                name="gender"
                onChange={[Function]}
                tabIndex={1}
                type="radio"
                value="Male" />,
              " ",
              "Male",
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "checked": false,
                "id": "Male",
                "name": "gender",
                "onChange": [Function],
                "tabIndex": 1,
                "type": "radio",
                "value": "Male",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            " ",
            "Male",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "Other",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                checked={false}
                id="Other"
                name="gender"
                onChange={[Function]}
                tabIndex={2}
                type="radio"
                value="Other" />,
              " ",
              "Other",
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "checked": false,
                "id": "Other",
                "name": "gender",
                "onChange": [Function],
                "tabIndex": 2,
                "type": "radio",
                "value": "Other",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            " ",
            "Other",
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
